### ISSUE TITLE
find --devices doesn't really order the result

### ISSUE THREAD

**Issue by liloman** [chenall/grub4dos#93]
Hi,

I have found an issue that's causing my troubles:

``` batch
grub>find --devices=hc /myfile.txt
(cd)
(hd0,1)
```

It's the same as: 

``` batch
grub>find --devices=ch /myfile.txt
(cd)
(hd0,1)
```

Cheers and nice piece of software :wave:

**Comment by liloman** [comment]
My bad for not looking first on the source.

Looking on the code is crystal clear that it's searching into the set-root device first. 

Shouldn't it respect the order of --devices?? 

Anyway I reckon I can fix my menus. ;)

**Comment by chenall** [comment]
because `find` will find current root first
so find --devices=ch /myfiles.txt
if `root` is (hd0,0),it find (hd0,0) first.

**Comment by liloman** [comment]
Yeah that is what I figured out so I patched it to not search for the current root first and problem solved.

By the way It will be nice to have this option. :smirk:

**Comment by steve6375** [comment]
maybe try

```
rootnoverify (rd)
find --devices=ch /myfiles.txt
```
